src/G2/Translation/Haskell.hs

```diff
@@ -250,22 +250,33 @@ mkCgGutsModDetailsClosures tr_con env modgutss = do
   tidys <- mapM (tidyProgram env) simplgutss
 #endif
 
-  let pairs = map (\((cg, md), mg) -> ( mkCgGutsClosure (mg_binds mg) cg
+  let pairs = map (\((cg, md), mg) -> ( mkCgGutsClosure (mg_binds mg) cg md
```

Comment:
Is line 253 and line 239 describe the same thing and I think I might need a comment on what pairs are used for?

Reply:
This function is (I think unavoidable) a bit messy because GHC has changed how it works several times, and we are currently trying to support GHC 8.2 - 9.6.  So lines 239 and 253 are the same, but notice we have C Pragmas controlling which get's executed based on the GHC version.

I added a comment to the function- really pairs is just information the function returns about stuff that it extracted from GHC.  If you're wondering something more specific, lmk.